src/globals/_variants.scss

```diff
@@ -0,0 +1,146 @@
+// Default style
+%variant_ghost {
+    color: var(--color-gray_dark);
+
+    border: 1px solid transparent;
+
+    background-color: transparent;
+
+    &:hover:not(:disabled) {
+        color: var(--color-accent_dark);
+
+        background-color: var(--color-gray_light_plain20);
+    }
+
+    &:focus-visible:not(:disabled) {
+        color: var(--color-accent_dark);
+
+        border-color: var(--color-accent_dark);
+        background-color: transparent;
+    }
+
+    &:active:not(:disabled) {
+        border-color: transparent;
+        background-color: var(--color-gray_light_plain60);
+    }
+
+    &.moonstone-reverse {
+        color: var(--color-light);
+
+        &:focus:not(:disabled),
+        &:hover:not(:disabled) {
+            color: var(--color-accent_light);
+
+            border-color: transparent;
+            background-color: var(--color-gray_light20);
+        }
+
+        &:active:not(:disabled) {
+            color: var(--color-white);
+
+            border-color: transparent;
+            background-color: var(--color-gray_light40);
+        }
+
+        &:disabled {
+            color: var(--color-gray_light);
+        }
+    }
+
+    &:disabled {
+        color: var(--color-gray_dark);
+
+        border-color: transparent;
+        background-color: transparent;
+        cursor: not-allowed;
+        opacity: 0.4;
+    }
+}
+
+%variant_ghost_accent {
+    @extend %variant_ghost;
```

Comment:
As `%variant_ghost` manages state `:hover`, `:focus` , `:active`. I don't think we need to extend it. Plus, it will almost be overridden.